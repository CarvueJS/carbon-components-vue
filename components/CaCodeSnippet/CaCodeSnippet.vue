<template>
  <div class="bx--snippet" :class="dynamicClass">
    <div class="bx--snippet-container">
      <code>
      <pre>
        <slot>
        </slot>
      </pre>
      </code>
    </div>
    <button data-copy-btn class="bx--snippet-button" aria-label="Copy code" tabindex="0">
      <svg class="bx--snippet__icon" width="18" height="24" viewBox="0 0 18 24" fill-rule="evenodd">
        <path d="M13 5V0H0v19h5v5h13V5h-5zM2 17V2h9v3H5v12H2zm14 5H7V7h9v15z"></path>
        <path d="M9 9h5v2H9zM9 12h5v2H9zM9 15h3v2H9z"></path>
      </svg>
      <div class="bx--btn--copy__feedback" data-feedback="Copied!"></div>
    </button>
  </div>

</template>

<script>
  export default {
    name: 'ca-code-snippet',
    props: {
      type: {
        type: String,
        default: 'code'
      }
    },
    computed: {
      dynamicClass () {
        return {
          'bx--snippet--terminal': this.type === 'terminal',
          'bx--snippet--code': this.type === 'code',
        }
      }
    },
  };
</script>

<style></style>
