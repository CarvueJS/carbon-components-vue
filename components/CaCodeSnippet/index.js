import CaCodeSnippet from './CaCodeSnippet.vue';

CaCodeSnippet.install = Vue => {
  Vue.component(CaCodeSnippet.name, CaCodeSnippet);
}

export default CaCodeSnippet
