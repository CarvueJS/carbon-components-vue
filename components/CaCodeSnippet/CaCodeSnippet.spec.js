import { shallow } from 'vue-test-utils';
import CaCodeSnippet from './CaCodeSnippet.vue';

describe('CaCodeSnippet', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaCodeSnippet);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
})
