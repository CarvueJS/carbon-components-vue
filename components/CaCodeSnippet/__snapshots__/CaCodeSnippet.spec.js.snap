// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaCodeSnippet Mathes snapshot 1`] = `
<div class="bx--snippet bx--snippet--code">
  <div class="bx--snippet-container"><code><pre>      
      
    </pre></code></div>
  <button data-copy-btn="" aria-label="Copy code" tabindex="0" class="bx--snippet-button">
    <svg width="18" height="24" viewBox="0 0 18 24" fill-rule="evenodd" class="bx--snippet__icon">
      <path d="M13 5V0H0v19h5v5h13V5h-5zM2 17V2h9v3H5v12H2zm14 5H7V7h9v15z"></path>
      <path d="M9 9h5v2H9zM9 12h5v2H9zM9 15h3v2H9z"></path>
    </svg>
    <div data-feedback="Copied!" class="bx--btn--copy__feedback"></div>
  </button>
</div>
`;
